NRI Webdev-Challenge

anandmsmaven opened the pull request:
Hello! My name is Anand Rajendran.

I have completed all the required task and few additional task as well and documented the setup on linux and also documented how to run the application in the wiki section

Hoping to hear from you

Thank you

anandmsmaven committed:
tdlr; Setting up the NRI Inventory System using Laravel 5 framework
Summary:
-- Using the composer downloaded the laravel framework 5.1 version
-- Modified the /etc/ hosts file to add hostname to IP address
-- Modified the homestead.yaml file to map hostname to the project folder
-- Run the vagrant up --provision to start the new application which we set to the hostname
-- Changed the .env.example to .env and added the DB name to connect the application with the Database
-- Created a new controller named HomeController.php and an Index method to load the welcome page
-- Changed the routes.php file to call the HomeController and Index method

anandmsmaven committed:
tdlr; Setting up the NRI Inventory System using Laravel 5 framework
Summary:
-- Using the composer downloaded the laravel framework 5.1 version
-- Modified the /etc/ hosts file to add hostname to IP address
-- Modified the homestead.yaml file to map hostname to the project folder
-- Run the vagrant up --provision to start the new application which we set to the hostname
-- Changed the .env.example to .env and added the DB name to connect the application with the Database
-- Created a new controller named HomeController.php and an Index method to load the welcome page
-- Changed the routes.php file to call the HomeController and Index method

anandmsmaven committed:
tdlr; Merging the data.csv and REadME.md files

anandmsmaven committed:
tdlr; An Inventory tool which allows to upload CSV file and
parse the data to store in the database and display the data
using the jquery datatables without page reload
Summary:
-- Added methods to upload the file and retrieve the data in Homecontroller.php
-- Added the routes to store and display the data in routes.app
-- Added a script named SimpleAjaxUploader.min.js to handle the file upload in public/js folder
-- Created a layout with navbar and file upload UI and datatable in home.blade.php
-- Added few points on the implementation that I'm proud of in README.md

anandmsmaven committed:
tdlr; An Inventory tool which allows to upload CSV file and
parse the data to store in the database and display the data
using the jquery datatables without page reload
Summary:
-- Added methods to upload the file and retrieve the data in Homecontroller.php
-- Added the routes to store and display the data in routes.app
-- Added a script named SimpleAjaxUploader.min.js to handle the file upload in public/js folder
-- Created a layout with navbar and file upload UI and datatable in home.blade.php
-- Added few points on the implementation that I'm proud of in README.md

anandmsmaven committed:
tdlr; Included the exported sql file in the project

anandmsmaven committed:
tdlr; Included the exported sql file in the project

anandmsmaven committed:
Merge branch 'master' of https://github.com/anandmsmaven/webdev-challenge

nrigroup commented:
Thank you for your submission.

I see that you mentioned in the controller as one of your todo list to move db related code into models. Would you mind doing that then re-submit for a pull request?

anandmsmaven committed:
tdlr; Implemented the complete MVC approach by registering the repository and interface in APPServiceProvider.php
Summary:
-- Created a DBHomeRepository and HomeRepositoryInterface in Repositories folder
-- Created methods for all the database activities in DBHomeRepository.php

anandmsmaven commented:
Moved the database related code to models registering the repositories and interfaces in the appservicefile.
Kindly take a look at the wiki section for the documentations

anandmsmaven commented:
Hello,

I have moved all the database queries to a repository. So the application
runs completely on MVC architecture.

Kindly take a look at the wiki section for documentation

Thank you

On Thu, Aug 17, 2017 at 2:48 PM, nrigroup <notifications@github.com> wrote:

> Thank you for your submission.
>
> I see that you mentioned in the controller as one of your todo list to
> move db related code into models. Would you mind doing that then re-submit
> for a pull request?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nrigroup/webdev-challenge/pull/27#issuecomment-323160911>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASma0-KkB4k7Pnz6rLZFZn4jNZ8e098Mks5sZIr2gaJpZM4O6i3e>
> .
>
